plotly/tools.py

```diff
@@ -51,6 +50,15 @@ def warning_on_one_line(message, category, filename, lineno,
 except ImportError:
     _numpy_imported = False
 
+try:
+    import scipy as scp
+    import scipy.spatial as scs 
+    import scipy.cluster.hierarchy as sch 
+
+    _scipy_imported = True
+except ImportError:
+    _scipy_imported = False
```

Comment:
Just as a preventative measure for conflicting optional import needs, would you mind incorporating a syntax as suggested in https://github.com/plotly/python-api/pull/275/files#r37724351 ?

@chriddyp does this seem reasonable? The other option I see is *tagging* import patterns as `_dendrogram_imports` or something... but that seems like it'd make the code harder to read?

It feels strange to me to force folks who have scipy and *aren't* using dendrograms to have to install that functionality... but maybe this is yet another viable route...